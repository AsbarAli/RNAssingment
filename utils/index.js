/* eslint-disable react/forbid-foreign-prop-types */
import checkPropTypes from 'check-prop-types';
import {applyMiddleware, createStore} from 'redux';

import PostReducer from '../src/module/reducers/PostReducer';
import {middleware} from '../src/store/ConfigureStore';

export const checkProps = (component, expectedProps) => {
  const propsError = checkPropTypes(component.propTypes, expectedProps, 'props', component.name);

  return propsError;
};

export const propsWithNavigation = (props: Object) => ({
  navigation: {
    navigate: jest.fn(),
  },
  ...props,
});

export const testStore = (INITIAL_STATE) => {
  const createdStoreMiddlewares = applyMiddleware(...middleware)(createStore);

  return createdStoreMiddlewares(PostReducer, INITIAL_STATE);
};
