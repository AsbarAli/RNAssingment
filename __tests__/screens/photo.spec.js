/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';
import {Provider} from 'react-redux';

import PhotoScreens from '../../src/module/screens/photo/Photo.screens';
import {testStore, propsWithNavigation} from '../../utils';

const store = testStore();

describe('Photo screen', () => {
  describe('Rendering', () => {
    let wrapper: ShallowWrapper;
    let props: any;

    beforeEach(() => {
      props = propsWithNavigation({});
      wrapper = shallow(
        <Provider store={store}>
          <PhotoScreens {...props} />
        </Provider>
      );
    });

    it('should match to snap shot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
