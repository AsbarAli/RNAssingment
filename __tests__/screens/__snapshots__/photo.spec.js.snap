// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photo screen Rendering should match to snap shot 1`] = `ShallowWrapper {}`;
